Update to OpenRA prep-2211

penev92 opened the pull request:
Updates the ModSDK to the latest OpenRA version, with the targeting of .NET 6 _(up from .NET Framework 4.7.2)_ and everything else.

It includes everything that was done for #171, https://github.com/pchote/OpenRAModSDK/pull/2 and https://github.com/pchote/TiberianDawnHD/pull/5.

Supersedes #171.

Will need an actual OpenRA release version to be added to `mod.config` once there is one.

Mailaender commented on mods/example/mod.yaml:
```diff
@@ -66,12 +67,18 @@ MapGrid:
 
 SpriteFormats: ShpTS
 
```
```suggestion

SoundFormats: Wav
```

as https://github.com/OpenRA/OpenRAModSDK/blob/e0c1ee07e4589335e92c8142e5f29530d4b4c6a5/mods/example/audio/notifications.yaml#L2

already mentions it.

penev92 replied:
Ok.

Mailaender commented:
Don't forget about https://github.com/OpenRA/OpenRA/pull/20610.

Mailaender commented:
Please integrate https://github.com/OpenHV/OpenHV/commit/9e89912b416dbbb7bc54846eb19da314edd130df which I forgot to upstream.

Mailaender commented on .github/workflows/packaging.yml:
```diff
@@ -9,10 +9,14 @@ jobs:
   linux:
     name: Linux AppImages
     runs-on: ubuntu-20.04
```
```suggestion
    runs-on: ubuntu-22.04
```

Mailaender replied:
and install `libfuse2`

Mailaender commented on .github/workflows/packaging.yml:
```diff
@@ -67,16 +75,20 @@ jobs:
   windows:
     name: Windows Installers
     runs-on: ubuntu-20.04
```
```suggestion
    runs-on: ubuntu-22.04
```

Mailaender commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,86 @@
+jobs:
+  linux:
+    name: Linux (.NET 6.0)
+    runs-on: ubuntu-20.04
```
```suggestion
    runs-on: ubuntu-22.04
```

Mailaender commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,86 @@
+
+  linux-mono:
+    name: Linux (mono)
+    runs-on: ubuntu-20.04
```
```suggestion
    runs-on: ubuntu-22.04
```

penev92 commented:
Updated GH Actions workflows to use Ubuntu 22, added `libfuse2` installation to the Linux packaging workflow, changed the Discrod variable and added the registry cleanup from https://github.com/OpenRA/OpenRA/pull/20610. Then I saw #181 already took care of that last part.
Thanks, @Mailaender!
